extensions/gadgets/ScoreBar/ScoreBar.js

```diff
@@ -21,21 +21,35 @@
  */
 
 oppia.directive('oppiaGadgetScoreBar', [
-  'oppiaHtmlEscaper', 'learnerParamsService', function(oppiaHtmlEscaper, learnerParamsService) {
-
+  'oppiaHtmlEscaper', 'learnerParamsService',
+  function(oppiaHtmlEscaper, learnerParamsService) {
     return {
       restrict: 'E',
       templateUrl: 'gadget/ScoreBar',
-      controller: ['$scope', '$attrs', function ($scope, $attrs) {
-
-        $scope.scoreBarLabel = oppiaHtmlEscaper.escapedJsonToObj($attrs.gadgetName);
-        $scope.maxValue = oppiaHtmlEscaper.escapedJsonToObj($attrs.maxValueWithValue);
-        $scope.scoreBarParamName = oppiaHtmlEscaper.escapedJsonToObj($attrs.paramNameWithValue);
+      controller: ['$scope', '$attrs', function($scope, $attrs) {
+        $scope.scoreBarLabel = oppiaHtmlEscaper.escapedJsonToObj(
+          $attrs.gadgetName);
+        $scope.maxValue = oppiaHtmlEscaper.escapedJsonToObj(
+          $attrs.maxValueWithValue);
+        $scope.scoreBarParamName = oppiaHtmlEscaper.escapedJsonToObj(
+          $attrs.paramNameWithValue);
 
         $scope.getScoreValue = function() {
-          return learnerParamsService.getValue($scope.scoreBarParamName);
+          // If a parameter's value exceeds the ScoreBar's maxValue, return
+          // maxValue to keep the label text visible.
+          if (learnerParamsService.getValue(
+              $scope.scoreBarParamName) > $scope.maxValue) {
+            return $scope.maxValue;
+          } else if (learnerParamsService.getValue(
+              $scope.scoreBarParamName) < 0) {
+            // Defense-in-depth: although underflows don't break the current
+            // UI, return a sensible minimum.
+            return 0;
+          } else {
+            return learnerParamsService.getValue($scope.scoreBarParamName);
+          }
```

Comment:
Actually -- we have a function for this, called 'clamp', here:

    https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/components/visualizations.js#L586

It might be worth creating a numberUtilsService and reusing that in both places. However, if you'd prefer not to do that at this stage, maybe at least make a separate clamp() method in this controller and use that instead, and we can unify the two clamp() methods later.